beacon-chain/rpc/beacon/committees.go

```diff
@@ -0,0 +1,151 @@
+package beacon
+
+import (
+	"context"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/pagination"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// ListBeaconCommittees for a given epoch.
+//
+// If no filter criteria is specified, the response returns
+// all beacon committees for the current epoch. The results are paginated by default.
+func (bs *Server) ListBeaconCommittees(
```

Comment:
For this we might need more options for the `req *ethpb.ListCommitteesRequest`. Along with epoch we also need to give a query option, where we can provide the state root. 

In the event of a fork in the network, you could have two chains with different committees at the same epoch. To differentiate on the two sets of committees we would need to be able to identify committees from forked chains, so the state root is a good way to identify them. @shayzluf and I discussed this offline, in the event of a fork in the network , we need to be able to get the appropriate committees from each forked chain.

Reply:
Great point, that's a neat suggestion. So what happens if you provide a state root that doesnt match head state?

Reply:
Maybe we can do this in a follow-up PR as that will increase the diff quite a lot.

Reply:
I guess it would mean we are asking for the committees from a forked chain :) This would be important for the slasher, as it needs to be able to detect malicious attestations in the event of a fork. Sounds good, we can do it in a follow up PR